### PR TITLE
EoCify and expand contact lenses

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Content "More contact lenses"
#### Purpose of change
It just happened
#### Describe the solution
Change currently existing contact lenses into multiple categories: daily wear / weekly disposable, myopia/hyperopia/bifocal/plano (only transition lenses), and default/transition lenses - 14 items in total
Put all new lenses into boxes, make them spawn in groups
Removed `iuse::contacts`, replaced with EoC that works almost the same way as old one
Added `has_flag_eval` to make the checks easier to read, thanks andrei8l for code
Added two events `character_attempt_to_fall_asleep` and `character_falls_asleep`, but then decided to not use it in this PR, see additional context
#### Todo
- [x] replicate "Replace your current lenses?" query that i forgot to finish
#### Testing
Lenses work as expected, you can find them in boxes as expected  but i should recompile and test it a bit more
#### Additional context
I tried to implement bi-weekly and monthly lenses initially, but rejected from them afterwards because they require constant supply of contact lens solution and container to store said lenses overnight, and i don't see a good way to implement it using EoC - i would like to hear some thoughts about it

### CODE REVIEW

**Review by nornagon — COMMENTED** [review]
copy editing. Also: "plano" doesn't mean anything to me (as a non-glasses-wearer), perhaps this should be replaced with "non-prescription" or "non-correcting"?

**Comment by GuardianDll** [comment]
\> "plano" doesn't mean anything to me

"plano" seems to be the official term, that describe this type of lenses, and i'd like to stick to it because it's shorter, but if it's mislead items in any way, then i'll change it for sure

File: data/json/effects.json [hunk 1]
```diff
@@ -3207,6 +3207,28 @@
...
+  {
+    "type": "effect_type",
+    "id": "transition_contacts_plano",
+    "name": [ "Transition Contact Lenses (Plano)" ],
+    "desc": [
+      "You are wearing a plano transition contact lenses, that protect you from the sunlight, but doesn't help with any sight issues."
+    ],
+    "blocks_effects": [ "glare" ],
+    "apply_message": "You can see more clearly.",
+    "remove_message": "Your vision starts to blur.",
```
Comment (GuardianDll) [inline data/json/effects.json:3229]:
```suggestion
    "apply_message": "You apply photochromic contact lenses.",
    "remove_message": "you pull off your photochromic contact lenses.",
```

File: data/json/items/comestibles/med.json [hunk 2]
```diff
@@ -469,36 +469,424 @@
...
+        ]
+      }
+    ]
+  },
+  {
+    "id": "myopia_transition_contacts_daily",
+    "type": "GENERIC",
+    "copy-from": "myopia_transition_contacts_weekly",
+    "name": { "str_sp": "myopia transition contact lenses (daily)" },
+    "description": "Small plastic case with two transition daily-use soft contacts lenses, buffered in saline solution.  They are a great replacement for wearing glasses and sit comfortably on the surface of the eye.  The lenses are designed to help anyone with <color_light_blue>nearsightedness</color>, and should be discarded after <color_light_blue>16 hours</color> of use.",
```
[the remark this answers is not in the document]
Comment (GuardianDll) [reply data/json/items/comestibles/med.json:577]:
```suggestion
    "description": "Small plastic case with two transition daily-use soft contact lenses, buffered in saline solution.  This high-tech pair darkens in response to light, making it easier to see in bright environments.  They are a great replacement for wearing glasses and sit comfortably on the surface of the eye.  The lenses are designed to help anyone with <color_light_blue>nearsightedness</color>, and should be discarded after <color_light_blue>16 hours</color> of use.",
```

File: data/json/items/comestibles/med.json [hunk 3]
```diff
@@ -469,36 +469,424 @@
...
+        ]
+      }
+    ]
+  },
+  {
+    "id": "bifocal_transition_contacts_weekly",
+    "type": "GENERIC",
+    "copy-from": "myopia_contacts_weekly",
+    "name": { "str_sp": "bifocal transition contact lenses (weekly)" },
+    "description": "Small plastic case with two transition extended-wear soft contacts lenses, buffered in saline solution.  They are a great replacement for wearing glasses and sit comfortably on the surface of the eye.  The lenses are designed to help anyone with <color_light_blue>both nearsightedness and farsightedness</color>, and should be discarded after <color_light_blue>a week</color> of use.",
```
[the remark this answers is not in the document]
Comment (GuardianDll) [reply data/json/items/comestibles/med.json:750]:
```suggestion
    "description": "Small plastic case with two transition extended-wear soft contact lenses, buffered in saline solution.  This high-tech pair darkens in response to light, making it easier to see in bright environments.  They are a great replacement for wearing glasses and sit comfortably on the surface of the eye.  The lenses are designed to help anyone with <color_light_blue>both nearsightedness and farsightedness</color>, and should be discarded after <color_light_blue>a week</color> of use.",
```

**Pull request merged**